src/managers/KeybindManager.cpp

```diff
@@ -2013,6 +2014,163 @@ void CKeybindManager::pass(std::string regexp) {
         g_pSeatManager->setPointerFocus(PWINDOW->m_pWLSurface.wlr(), SL);
 }
 
+void CKeybindManager::sendshortcut(std::string args) {
+
+    //args=<NEW_MODKEYS><NEW_KEY>[,WINDOW_RULES]
+    const auto ARGS = CVarList(args, 3);
+    if (ARGS.size() != 3) {
+        Debug::log(ERR, "sendshortcut: invalid args");
+        return;
+    }
+
+    const auto MOD     = g_pKeybindManager->stringToModMask(ARGS[0]);
+    const auto KEY     = ARGS[1];
+    uint32_t   keycode = 0;
+    bool       isMouse = 0;
+
+    //oriented from parseKey in ConfigManager
+    if (isNumber(KEY) && std::stoi(KEY) > 9)
+        keycode = std::stoi(KEY);
+    else if (KEY.compare(0, 5, "code:") == 0 && isNumber(KEY.substr(5)))
+        keycode = std::stoi(KEY.substr(5));
+    else if (KEY.compare(0, 6, "mouse:") == 0 && isNumber(KEY.substr(6))) {
+        keycode = std::stoi(KEY.substr(6));
+        isMouse = 1;
+        if (keycode < 272) {
+            Debug::log(ERR, "sendshortcut: invalid mouse button");
+            return;
+        }
+    } else {
+
+        // here, we need to find the keycode from the key name
+        // this is not possible through xkb's lib, so we need to iterate through all keycodes
+        // once found, we save it to the cache
+
+        const auto KEYSYM = xkb_keysym_from_name(KEY.c_str(), XKB_KEYSYM_CASE_INSENSITIVE);
+        keycode           = 0;
+
+        const auto KB = g_pSeatManager->keyboard;
+
+        if (!KB) {
+            Debug::log(ERR, "sendshortcut: no kb");
+            return;
+        }
+
+        // create a unique string for the keypair
+        // casting wlr_keyboard* to void* for formatting
+        const auto KEYPAIRSTRING = std::format("{:p}{}", static_cast<void*>(KB->wlr()), KEY);
```

Comment:
generally we do `{}` with `(uintptr_t)var.get()`

Reply:
Done